src/kbmod_wf/ic_workflow.py

```diff
@@ -0,0 +1,260 @@
+import argparse
+import os
+import toml
+import parsl
+from parsl import python_app, File
+import parsl.executors
+
+from kbmod_wf.utilities.configuration_utilities import apply_runtime_updates, get_resource_config
+from kbmod_wf.utilities.executor_utilities import get_executors
+from kbmod_wf.utilities.logger_utilities import configure_logger
+
```

Comment:
Just a naive parsl question: how do files get outputted where it is clear what workflow is used? Is `ic_workflow` used in the output path?

If this workflow will be chip-by-chip specific, it might be worth naming this workflow file to reflect that (especially if the above is true)

Reply:
This is a good observation, and one that I could use a little help with. Currently all that happens is that the intermediate and final files are output to the staging directory. 

Open simple option would be to add a subdirectory `.../staging/ic_workflow/<output_files>`.

Beyond that I think that there would need to be a little more flexibility - and this is where I could really use some help. If multiple people are pulling from the staging directory, then we would need a configurable way to identify the output location such that we wouldn't clobber each others results _and_ wouldn't accidentally duplicate work - i.e. pull files out of the staging area when they are being processed?

Reply:
Yeah I agree with a subdirectory being helpful.

Allowing multiple concurrent users of the staging directory adds some complications. Pulling files out of the staging directory sounds like a small waste of resources since you would still want the file written out somewhere else in case of catastrophic failure. Maybe a shared in-progress manifest file with a lock file? Then when someone starts a run 

1. gets list of files from staging area
2. acquires global manifest lock
3. subtracts in-progress files from the global manifest and adds any new files to process
4. releases global manifest lock
5. writes its own unique manifest file for that run?
6. If catchable failure, acquires global manifest lock and removes files that weren't processed? (hard to figure out across multiple tasks?)

(6) feels convoluted enough to suggest using a more standardized queueing system rather than rolling our own

A good issue but probably out of scope for this PR

Reply:
I have an issue here, https://github.com/dirac-institute/kbmod-wf/issues/25, that partially gets at this. I'll add a link to your comment for record keeping. But I agree, a final solution to this problem is outside the scope of this PR.

Reply:
I'll implement the simple solution first of simply outputing data to `.../staging/<workflow_name>/` for this PR and pick up the rest of the work next.

Reply:
I've addressed this by including a new configuration parameter that will specify an `output_directory` where the results will be stored. Again, for the time being, we can just manually set this to be the workflow name, as in `.../staging/<workflow_name>/`.

Reply:
Cool!